engine/service/service.go

```diff
@@ -115,11 +115,24 @@ func (c *Common) Register(ctx context.Context, cfg sdk.ServiceConfig) error {
 		srv.PublicKey = pubKeyPEM
 	}
 
-	srv2, err := c.Client.ServiceRegister(ctx, srv)
-	if err != nil {
-		return sdk.WrapError(err, "Register>")
+	retry := 0
+	for {
+		srv2, err := c.Client.ServiceRegister(ctx, srv)
```

Comment:
I think the hatchery must be able to register without CDN.
the hatchery should get the CDN information (like very minutes)
But the hatchery should not try start worker if there is no CDN information.
If CDN is unregistered, the hatchery should not flap but raise alert and avoid to start new worker